library/ssl_tls13_client.c

```diff
@@ -176,10 +176,13 @@ int ssl_write_early_data_process( mbedtls_ssl_context* ssl )
 #endif /* MBEDTLS_SSL_USE_MPS */
 
 #else /* MBEDTLS_ZERO_RTT */
+#if defined(MBEDTLS_SSL_USE_MPS)
```

Comment:
ISTM that we should rather guard the declarations of `buf, buf_len, ...` by `MBEDTLS_SSL_USE_MPS && MBEDTLS_ZERO_RTT`.